### PR TITLE
WellKnownNames

### PR DESCRIPTION
**Pull request by jansule**
**Important:** Please wait with merging this PR until

- [x] [this PR from geostyler-style](https://github.com/terrestris/geostyler-style/pull/43)
- [x] [this PR from geostyler-sld-parser](https://github.com/terrestris/geostyler-sld-parser/pull/62)
- [x] [this PR from geostyler-openlayers-parser](https://github.com/terrestris/geostyler-openlayers-parser/pull/20)
were merged.

Handling WellKnownNames for PointSymbolizers in GUI. Removed `Circle` from `KindField` in Symbolizer and replaced it with `Mark`. Added three new components:
- **MarkEditor** - Handles changes in WellKnownNameField and WellKnownNameEditor
- **WellKnownNameEditor** - Handles changes of properties of a single WellKnownName
- **WellKnownNameField** - Dropdown menu for switching between supported WellKnownNames

Supported WellKnownNames are `Circle`, `Square`, `Triangle`, `Star`, `Cross` and `X`.

CircleEditor was removed since new components also handle Circles.

All Symbols can be rotated.

**Example 1:** Default WellKnownName `Circle`.
image

**Example 2:** WellKnownName `Square`:
image

**Example 3:** WellKnownName `Triangle`:
image

**Example 4:** WellKnownName `Star`:
image

**Example 5:** WellKnownName `Cross`:
image

**Example 6:** WellKnownName `X`:
image

### CODE REVIEW

File: src/Component/Style/Style.tsx [hunk 1]
```diff
@@ -157,7 +129,7 @@ class Style extends React.Component<StyleProps, StyleState> {
     const randomId = Math.floor(Math.random() * 10000);
     const newRule: GsRule = {
       name: 'rule_' + randomId,
-      symbolizer: this.getSymbolizerFromStyleType(style)
+      symbolizer: [this.getDefaultSymbolizer(style.rules[0].symbolizer[0])]
```
Comment (dnlkoch) [inline src/Component/Style/Style.tsx]: Is `style.rules[0].symbolizer[0]` available always?

Comment (jansule) [reply src/Component/Style/Style.tsx]: Not necessarily. Thanks for pointing out. Now using lodash's `get` function at this point and add handling if parameter in `getDefaultSymbolizer()` is undefined. See [this commit](https://github.com/terrestris/geostyler/pull/337/commits/efe89067b90763238668783c16f0f52e81b1aaed).

File: src/locale/de_DE.js [hunk 2]
```diff
@@ -101,5 +103,16 @@ export default {
...
+          Circle: 'Kreis',
+          Square: 'Quadrat',
+          Triangle: 'Dreieck',
+          Star: 'Stern',
+          Cross: 'Kreuz',
+          X: 'X'
+        }
+    },
     ...de_DE
 };
```
Comment (marcjansen) [inline src/locale/de_DE.js]: missing end of file newlines. super minor, of course.

File: src/Component/Symbolizer/Field/WellKnownNameField/WellKnownNameField.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  };
+
+  getWKNSelectOptions = (locale: WellKnownNameFieldLocale) => {
+    return this.props.wellKnownNames!.map(name => {
+      return (
+        <Option
+          key={name}
+          value={name}
+        >
+          {locale.wellKnownNames[name]}
```
Comment (dnlkoch) [inline src/Component/Symbolizer/Field/WellKnownNameField/WellKnownNameField.tsx]: Check for NP needed?

File: src/Component/Symbolizer/Field/WellKnownNameField/WellKnownNameField.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    Cross: string;
+    X: string;
+  };
+}
+
+// default props
+interface WellKnownNameFieldDefaultProps {
+  wellKnownName: WellKnownName;
+  wellKnownNames: WellKnownName[];
+  locale?: WellKnownNameFieldLocale;
```
Comment (dnlkoch) [inline src/Component/Symbolizer/Field/WellKnownNameField/WellKnownNameField.tsx]: Add default language prop?

**Comment by jansule** [comment]
@terrestris/devs ready to merge.
Now, it should be possible to remove all rules and all symbolizers without crashing.

**Comment by annarieger** [comment]
Thanks, merging :pineapple:

**Pull request merged**